telegram/ext/filters.py

```diff
@@ -2545,3 +2546,60 @@ def filter(self, message: Message) -> bool:
 
 VOICE = _Voice("filters.VOICE")
 """Messages that contain :attr:`telegram.Message.voice`."""
+
+
+class Mention(MessageFilter):
+    """Mention messages. If list of username or chat_id or :attr:`telegram.User` objects passed,
+    it filters messages to only allow those whose contains one of username
+    or chat_id or :attr:`telegram.User`.
+    Note:
+        Usernames only allowed with @ prefix. Example: "@username"
+        Chat id only allowed with positive numbers. Example: 123456
+    Examples:
+        A simple use case for passing username is to allow only messages that were sent with
+        mention username:
+        MessageHandler(filters.Mention("@username"), callback_method)
+        MessageHandler(filters.Mention(["@username",123456, `user` ), callback_method)
+    Args:
+        mentions (Union[:obj:`str` ,:obj:`str` ,:attr:`telegram.User` ] ||
+        Collection[Union[:obj:`str` ,:obj`str` ,:attr:`telegram.User` ]]):
+        Which messages to allow with username or chat_id or :attr:`telegram.User` object.
+        If one of them matched it will allow. If not specified,
+        will allow any text message with mention.
+    """
+
+    __slots__ = ("mention", "_mentions")
+
+    def __init__(self, mentions: SCT[Union[int, str, TGUser]]):
+        self._mentions = self._parse_mentions(mentions)
+        super().__init__(name=f"filters.Mention({mentions})")
+
+    @staticmethod
+    def _parse_mentions(mentions: SCT[Union[int, str, TGUser]]) -> Set[Union[int, str, TGUser]]:
+        if isinstance(mentions, Iterable) and not isinstance(mentions, str):
+            return set(mentions)
+        return {mentions}
+
+    @property
+    def mentions(self) -> FrozenSet[Union[int, str, TGUser]]:
+        """Which mention(s) to allow through.
+        Returns:
+            frozenset(Union[:obj:`int` , :obj:`str`, :attr:`telegram.User` ])
+        """
+        return frozenset(self._mentions)
+
+    @staticmethod
+    def _check_mention(message: Message, mention: Union[int, str, TGUser]) -> bool:
+        if isinstance(mention, TGUser):
+            return bool(any(mention.id == e.user.id for e in message.entities if e.user))
+        if isinstance(mention, int):
+            return bool(any(mention == e.user.id for e in message.entities if e.user))
+        return bool(
+            isinstance(mention, str)
+            and bool(any(e.type == MessageEntity.MENTION for e in message.parse_entities()))
```

Comment:
Please have a close look at the docs of `Message.parse_entities()` to see what it returns. Hint: It's not a list of message entities ;)

Reply:
Oh, After using in real example I understand you, and fixed this issue